File: php/test/QueryTests.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2008-2011, Sebastian Staudt
 *
 * @license http://www.opensource.org/licenses/bsd-license.php New BSD License
 */

error_reporting(E_ALL & ~E_USER_NOTICE);

require_once dirname(__FILE__) . '/../lib/steam-condenser.php';

require_once 'PHPUnit/Framework.php';

/**
 * @author     Sebastian Staudt
 * @package    steam-condenser
 * @subpackage tests
 */
class QueryTests extends PHPUnit_Framework_TestCase {

    /**
     * @expectedException TimeoutException
     */
    public function testInvalidGoldSrcServer() {
        $server = new GoldSrcServer(new InetAddress('1.0.0.0'), 27015);
        $server->getPing();
    }

    /**
     * @expectedException TimeoutException
     */
    public function testInvalidSourceServer() {
        $server = new SourceServer(new InetAddress('1.0.0.0'), 27015);
        $server->getPing();
    }

    public function testRandomGoldsrcServer() {
        $masterServer = new MasterServer(MasterServer::GOLDSRC_MASTER_SERVER);
        $serverArray = $masterServer->getServers(MasterServer::REGION_ALL, '\type\d\empty\1\full\1\gamedir\valve');
        $serverAddress = $serverArray[array_rand($serverArray)];
        $server = new GoldSrcServer(new InetAddress($serverAddress[0]), $serverAddress[1]);
        $server->initialize();
        $server->updatePlayerInfo();
        $server->updateRulesInfo();

        $this->assertNotNull($server->getPing());
        $this->assertNotNull($server->getPlayers());
        $this->assertNotNull($server->getRules());
        $this->assertNotNull($server->getServerInfo());

        echo $server;
    }

    public function testRandomSourceServer() {
        $masterServer = new MasterServer(MasterServer::SOURCE_MASTER_SERVER);
        $serverArray = $masterServer->getServers(MasterServer::REGION_ALL, '\type\d\empty\1\full\1\gamedir\tf');
        $serverAddress = $serverArray[array_rand($serverArray)];
        $server = new SourceServer(new InetAddress($serverAddress[0]), $serverAddress[1]);
        $server->initialize();
        $server->updatePlayerInfo();
        $server->updateRulesInfo();

        $this->assertNotNull($server->getPing());
        $this->assertNotNull($server->getPlayers());
        $this->assertNotNull($server->getRules());
        $this->assertNotNull($server->getServerInfo());

        echo $server;
    }
}

File: ruby/test/query_tests.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2008-2011, Sebastian Staudt

require 'ipaddr'
require 'test/unit'

$:.push File.join(File.dirname(__FILE__), '..', 'lib')

require 'errors/timeout_error'
require 'steam/servers/goldsrc_server'
require 'steam/servers/master_server'
require 'steam/servers/source_server'

class QueryTests < Test::Unit::TestCase

  # This test tries to initialize an invalid GoldSrc server
  def test_invalid_goldsrc_server
    assert_raise SteamCondenser::TimeoutError do
      invalid_server = GoldSrcServer.new IPAddr.new('1.0.0.0')
      invalid_server.ping
    end
  end

  # This test tries to initialize an invalid Source server
  def test_invalid_source_server
    assert_raise SteamCondenser::TimeoutError do
      invalid_server = SourceServer.new IPAddr.new('1.0.0.0')
      invalid_server.ping
    end
  end

  # This test gets a random GoldSrc server from the master server and does a
  # full query on it
  def test_random_goldsrc_server
    assert_nothing_raised do
      master_server = MasterServer.new(*MasterServer::GOLDSRC_MASTER_SERVER)
      servers = master_server.servers MasterServer::REGION_ALL, '\type\d\empty\1\full\1\gamedir\valve'

      assert !servers.empty?, 'Got no servers from master server.'

      server = GoldSrcServer.new(*servers[rand(servers.length)])
      server.init
      server.update_players
      server.update_rules

      print server.to_s
    end
  end

  # This test gets a random Source server from the master server and does a
  # full query on it
  def test_random_source_server
    assert_nothing_raised do
      master_server = MasterServer.new(*MasterServer::SOURCE_MASTER_SERVER)
      servers = master_server.servers MasterServer::REGION_ALL, '\type\d\empty\1\full\1\gamedir\tf'

      assert !servers.empty?, 'Got no servers from master server.'

      server = SourceServer.new(*servers[rand(servers.length)])
      server.init
      server.update_players
      server.update_rules

      print server.to_s
    end
  end

end
